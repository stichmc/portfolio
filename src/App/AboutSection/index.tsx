import { FileUser, Github, Linkedin, Mails } from "lucide-react";
import { div as MotionDiv } from "motion/react-client";

import env from "@/env";
import useIsMobileView from "@/hooks/useIsMobileView";
import FadeUp from "@/components/FadeUp";
import { Button } from "@/components-shadcn/button";
import { ButtonGroup } from "@/components-shadcn/button-group";

import HeadShotDisplay from "./HeadShotDisplay";

export interface AboutSectionProps {
  headShotImage: string;
}

const AboutSection = ({ headShotImage }: AboutSectionProps) => {
  const isMobileView = useIsMobileView();

  return (
    <div className="flex flex-col-reverse lg:grid lg:grid-cols-2 md:gap-12 gap-8 xl:gap-24 items-center pt-24 sm:pt-0 justify-end sm:justify-center h-[80vh] max-w-[1500px] sm:p-12 p-4">
      <div className="flex flex-col items-center gap-4 justify-center text-center">
        <FadeUp duration={0.5}>
          <h1 className="scroll-m-20 text-center text-2xl md:text-3xl xl:text-4xl font-extrabold text-balance">
            {!isMobileView && (
              <MotionDiv
                style={{ display: "inline-block", paddingRight: 12 }}
                animate={{
                  rotate: [0, 14, -8, 14, -4, 10, 0],
                }}
                transition={{
                  duration: 2.5,
                  ease: "easeInOut",
                  repeat: Infinity,
                }}
              >
                ðŸ‘‹
              </MotionDiv>
            )}
            Hi, I'm Michael Stich
          </h1>
        </FadeUp>
        <FadeUp delay={0.75} duration={0.5} className="max-w-[500px]">
          <h2 className="scroll-m-20 text-lg md:text-xl xl:text-2xl font-semibold tracking-tight first:mt-0">
            A software engineer with 3 years of professional work experience.
          </h2>
        </FadeUp>
        <FadeUp delay={1.5} duration={0.5} className="max-w-[550px]">
          {!isMobileView && (
            <ButtonGroup>
              <Button
                variant="outline"
                className="flex items-center justify-center gap-2 text-lg"
                onClick={() => window.open("https://linkedin.com/in/mcstich/", "_blank")}
              >
                <Linkedin /> LinkedIn
              </Button>

              <Button
                variant="outline"
                className="flex items-center justify-center gap-2 text-lg"
                onClick={() => window.open("https://github.com/stichmc/", "_blank")}
              >
                <Github /> Github
              </Button>

              <Button
                variant="outline"
                className="flex items-center justify-center gap-2 text-lg"
                onClick={() => window.open(`${env.BASE_URL}Michael_Stich_Resume.pdf`, "_blank")}
              >
                <FileUser /> Resume
              </Button>

              <Button
                variant="outline"
                className="flex items-center justify-center gap-2 text-lg"
                onClick={() => window.open("mailto:mcstich@outlook.com")}
              >
                <Mails /> Email
              </Button>

              {/* <Button
              variant="outline"
              className="flex items-center justify-center gap-2 text-lg"
              onClick={() => window.open("https://github.com/stichmc/portfolio/", "_blank")}
            >
              <Code /> Source Code
            </Button> */}
            </ButtonGroup>
          )}

          {isMobileView && (
            <div className="flex flex-row gap-4 items-center justify-center w-full">
              <ButtonGroup orientation="vertical">
                <Button
                  variant="outline"
                  className="flex items-center justify-center gap-2"
                  onClick={() => window.open("https://linkedin.com/in/mcstich/", "_blank")}
                >
                  <Linkedin /> LinkedIn
                </Button>

                <Button
                  variant="outline"
                  className="flex items-center justify-center gap-2"
                  onClick={() => window.open("https://github.com/stichmc/", "_blank")}
                >
                  <Github /> Github
                </Button>
              </ButtonGroup>

              <ButtonGroup orientation="vertical">
                <Button
                  variant="outline"
                  className="flex items-center justify-center gap-2"
                  onClick={() => window.open(`${env.BASE_URL}Michael_Stich_Resume.pdf`, "_blank")}
                >
                  <FileUser /> Resume
                </Button>

                <Button
                  variant="outline"
                  className="flex items-center justify-center gap-2"
                  onClick={() => window.open("mailto:mcstich@outlook.com")}
                >
                  <Mails /> Email
                </Button>
              </ButtonGroup>
            </div>
          )}
        </FadeUp>
        <FadeUp delay={1.5} duration={0.5} className="max-w-[550px]">
          <h3 className="leading-6 text-xs md:text-sm xl:text-base">
            I build and scale applications for large user bases across the entire stack: frontend, backend, and cloud.
            I'm very passionate about solving complex problems and working with extraordinary people.
          </h3>
        </FadeUp>
      </div>

      <div className="flex items-center justify-center">
        <HeadShotDisplay headShotImage={headShotImage} />
      </div>
    </div>
  );
};

export default AboutSection;
