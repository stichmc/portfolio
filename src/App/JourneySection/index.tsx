import FadeUp from "@/components/FadeUp";
import TimeLine from "./Timeline";

const JourneySection = () => {
  return (
    <FadeUp delay={2} duration={0.5}>
      <div className="w-full h-full flex items-center justify-start flex-col">
        <div className="scroll-m-20 border-b-2 border-card-foreground pb-2 text-xl md:text-2xl xl:text-3xl font-bold w-full xl:w-3/4 text-center">
          My Journey
        </div>
        <TimeLine />
      </div>
    </FadeUp>
  );
};

export default JourneySection;
