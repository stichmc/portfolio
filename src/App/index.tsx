import { useState, useEffect } from "react";
import { Loader } from "lucide-react";

import headShotImage from "@/assets/headshot.jpg";

import AboutSection from "./AboutSection";
import JourneySection from "./JourneySection";

const App = () => {
  const [imageLoaded, setImageLoaded] = useState(false);

  useEffect(() => {
    const img = new Image();
    img.src = headShotImage;
    img.onload = () => {
      setImageLoaded(true);
    };
  }, []);

  if (!imageLoaded) {
    return (
      <div className="h-screen flex justify-center items-center">
        <Loader className="animate-spin h-32 w-32" />
      </div>
    );
  }

  return (
    <div className="flex flex-col items-center justify-start">
      <AboutSection headShotImage={headShotImage} />

      <JourneySection />
    </div>
  );
};

export default App;
