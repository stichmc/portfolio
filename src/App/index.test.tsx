import { expect, test, beforeAll, vi } from "vitest";
import { render } from "@testing-library/react";
import App from ".";

beforeAll(() => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: vi.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: vi.fn(),
      removeListener: vi.fn(),
      addEventListener: vi.fn(),
      removeEventListener: vi.fn(),
      dispatchEvent: vi.fn(),
    })),
  });
});

test("App", () => {
  const rendering = render(<App />);
  expect(rendering).toMatchSnapshot();
});
