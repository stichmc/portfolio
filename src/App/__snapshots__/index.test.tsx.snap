// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen flex justify-center items-center"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-loader animate-spin h-32 w-32"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2v4"
          />
          <path
            d="m16.2 7.8 2.9-2.9"
          />
          <path
            d="M18 12h4"
          />
          <path
            d="m16.2 16.2 2.9 2.9"
          />
          <path
            d="M12 18v4"
          />
          <path
            d="m4.9 19.1 2.9-2.9"
          />
          <path
            d="M2 12h4"
          />
          <path
            d="m4.9 4.9 2.9 2.9"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen flex justify-center items-center"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-loader animate-spin h-32 w-32"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2v4"
        />
        <path
          d="m16.2 7.8 2.9-2.9"
        />
        <path
          d="M18 12h4"
        />
        <path
          d="m16.2 16.2 2.9 2.9"
        />
        <path
          d="M12 18v4"
        />
        <path
          d="m4.9 19.1 2.9-2.9"
        />
        <path
          d="M2 12h4"
        />
        <path
          d="m4.9 4.9 2.9 2.9"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
